src/map/skill.cpp

```diff
@@ -8717,16 +8717,17 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 
 			if (sd && sd->sc.data[SC_PETROLOGY_OPTION])
 				brate = sd->sc.data[SC_PETROLOGY_OPTION]->val3;
+			// If the target has SC_STONEWAIT or SC_STONE, cancel at 100%
+			if (tsc->data[type] || tsc->data[SC_STONE])
+				brate = 100;
```

Comment:
I thought we wanted to use EndReturn for this?

Also, couldn't it still be resisted even if we add 100% success chance?

Reply:
Agree and should not be needed as rate should be checked after failing and endreturn.